### PR TITLE
Fixes #23373 - Fix host dissociation when updating host

### PR DESCRIPTION
**Pull request by shiramax**
Host gets disassociate from the VM when the following happens:
1. Creating the host with a certain "hostgroup" and  changing the "Deploy on" from the inherited value to something else.
2. After the host created -> Edit the host 

This happens because in Update, the "Deploy on" is disabled (it shouldnt be possible to edit the compute resource), and this means that the form doesn't send the value of "Deploy on", but it does send the values of selected hostgroup. So the the "Deploy on" gets overriten by the "hostgroup".

The solution is to simply delete the "compute_resource_id"  parameter when updating the host,  there is no problem to do so because it's not possible to edit the compute resource after the host created.

### CODE REVIEW

File: app/controllers/api/v2/hosts_controller.rb [hunk 1]
```diff
@@ -156,6 +156,7 @@ def update
         @all_parameters = true
 
         @host.attributes = host_attributes(host_params, @host)
+        @host.attributes.delete(:compute_profile_id)
```
Comment (iNecas) [inline app/controllers/api/v2/hosts_controller.rb:159]: Is this related?

**Comment by iNecas** [comment]
I can confirm `attributes.delete(:compute_resource_id)` fixed the issue for me

**Comment by iNecas** [comment]
it would be nice to get some integration test for this one. Should not be hard to do (see https://github.com/theforeman/foreman/blob/develop/test/integration/host_js_test.rb#L183) and could be more confident this will keep working in the future.

**Review by dLobatog — APPROVED** [review]
Thanks @shiramax !

**Comment by ares** [comment]
@shiramax would you mind to send a cp to 1.19-stable and 1.18-stable branches? the redmine issue target version is currently set to 1.18.1
